frontend/src/components/proof-templates/AttributeGroup.vue

```diff
@@ -240,38 +247,36 @@ export default {
     },
   },
   methods: {
-    getIssuerLabel(restriction) {
+    getIssuerLabel(restriction: SchemaRestrictions) {
       const filteredIssuers = this.schemaTrustedIssuers.find(
-        (s) => s.issuerDid === restriction.issuerDid
+        (s: TrustedIssuer) => s.issuerDid === restriction.issuerDid
       );
 
       return filteredIssuers !== undefined &&
         Object.prototype.hasOwnProperty.call(filteredIssuers, "label")
         ? filteredIssuers.label
         : "";
     },
-    renderSchemaLabelId(attributeGroup) {
-      // FIXME: This needs refactoring
-      // This tries to show a schema and label but will show the attribute group if
-      let schemaId;
-      let internalSchemaId;
+    renderSchemaLabelId(attributeGroup: AttributeGroup) {
+      let schemaId: string;
+      let internalSchemaId: string;
       if (attributeGroup.schemaId) {
         internalSchemaId = attributeGroup.schemaId;
       } else if (
         attributeGroup.schemaLevelRestrictions &&
-        attributeGroup.schemaLevelRestrictions.schemaId
+        attributeGroup.schemaLevelRestrictions[0].schemaId
       ) {
-        schemaId = attributeGroup.schemaLevelRestrictions.schemaId;
+        schemaId = attributeGroup.schemaLevelRestrictions[0].schemaId;
```

Comment:
same here, if this is the frontend part and this is always set ignore the comment. I just know that in the backend this can be null.

Reply:
See above